node/src/components/contract_runtime.rs

```diff
@@ -386,16 +387,37 @@ impl ContractRuntime {
             }
             ContractRuntimeRequest::GetAddressableEntity {
                 state_root_hash,
-                key,
+                entity_addr,
                 responder,
             } => {
                 trace!(?state_root_hash, "get addressable entity");
                 let metrics = Arc::clone(&self.metrics);
                 let data_access_layer = Arc::clone(&self.data_access_layer);
                 async move {
                     let start = Instant::now();
-                    let request = AddressableEntityRequest::new(state_root_hash, key);
+                    let entity_key = match entity_addr {
+                        EntityAddr::SmartContract(_)|EntityAddr::System(_) => Key::AddressableEntity(entity_addr),
+                        EntityAddr::Account(account) => Key::Account(AccountHash::new(account)),
```

Comment:
I guess this relies on the fact that `AddressableEntityRequest` supports retrieving entities for legacy AccountHash? Might be worth leaving a comment about that because it's not obvious here

Reply:
The treatment of the `Account` is not changed in this PR, ie.:
- Previously we used `Key::from<AccountHash>` to produce `Key::Account` and this was put into `AddressableEntityRequest::new`
- Now we create `EntityAddr::Account(<bytes>)`, then in Contract Runtime we get these bytes and create `AccountHash` out of them and then we pass this to `Key::Account`

Hence, while there should be no regression introduced by this PR, we should double check if the original code isn't buggy and cover this with test in a separate ticket/PR.